utils/general.py

```diff
@@ -36,6 +38,73 @@
 # Prevent OpenCV from multithreading (to use PyTorch DataLoader)
 cv2.setNumThreads(0)
 
+# Download objects
+def download_file(dir, blob):
+    file_name = blob.name.split("/")[2]
+    destination = Path(dir) / f"{file_name}"
+    destination.touch()
+    blob.download_to_filename(destination)
+
+
+# Place GCS objects in their respective local directories
+def place_object(local_dir, blob):
+    image_train = "images/t"
+    image_val = "images/v"
+    label_train = "labels/t"
+    label_val = "labels/v"
+    jpg = ".jpg"
+    txt = ".txt"
+
+    if blob.name.startswith(image_train) and blob.name.endswith(jpg):
+        download_file(local_dir[0], blob)
+
+    elif blob.name.startswith(label_train) and blob.name.endswith(txt):
+        download_file(local_dir[1], blob)
+
+    elif blob.name.startswith(image_val) and blob.name.endswith(jpg):
+        download_file(local_dir[2], blob)
+
+    elif blob.name.startswith(label_val) and blob.name.endswith(txt):
+        download_file(local_dir[3], blob)
+
+    else:
+        pass
+
+ 
+# Decorator for "check_dataset"
+def check_decorator(func):
```

Comment:
I would also add an argument to the decorator, instead of hardcoding the `data_dict["bucket"]` attribute inside the function.

Then we can use the decorator more generally, if we need that.
```python
@check_decorator(bucket_name)
def some_function():
    ...
```

Reply:
I might need some clarification on this ;)